packages/react-search-ui/src/SearchProvider.js

```diff
@@ -10,29 +10,46 @@ import defaultA11yMessages from "./A11yNotifications";
  * The SearchProvider primarily holds a reference to the SearchDriver and
  * exposes it to the rest of the application in a Context.
  */
-const SearchProvider = ({ children, config }) => {
-  const [driver, setDriver] = useState(null);
+const SearchProvider = ({ children, config = {}, driver }) => {
+  const [driverInstance, setDriverInstance] = useState(null);
 
   useEffect(() => {
     // This initialization is done inside of useEffect, because initializing the SearchDriver server side
     // will error out, since the driver depends on window. Placing the initialization inside of useEffect
     // assures that it won't attempt to initialize server side.
-    const newDriver = new SearchDriver({
-      ...config,
-      a11yNotificationMessages: {
-        ...defaultA11yMessages,
-        ...config.a11yNotificationMessages
-      }
-    });
-    setDriver(newDriver);
+    const currentDriver =
+      driver ||
+      new SearchDriver({
+        ...config,
+        a11yNotificationMessages: {
+          ...defaultA11yMessages,
+          ...config.a11yNotificationMessages
+        }
+      });
+    setDriverInstance(currentDriver);
+
     return () => {
-      newDriver.tearDown();
+      currentDriver.tearDown();
     };
   }, []);
 
+  // This effect allows users to dynamically update their searchQuery without re-mounting a SearchProvider,
+  // which would be destructive. An example of why this is useful is dynamically updating facets.
+  useEffect(() => {
```

Comment:
This effect (and the next effect) are meant to only run after the initial mount. So this is basically equivalent to `componentDidUpdate`. The first time it runs on mount it will do nothing since `driverInstance` will be null.